runtime/src/main/scala/com/soundcloud/twinagle/ServerBuilder.scala

```diff
@@ -0,0 +1,32 @@
+package com.soundcloud.twinagle
+
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.finagle.{Filter, Service}
+import com.twitter.util.Future
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion, Message}
+
+case class ServerBuilder(
+    extension: EndpointMetadata => Filter.TypeAgnostic,
+    endpoints: Map[EndpointMetadata, Service[Request, Response]] = Map.empty
+) {
+
+  def register[
```

Comment:
I considered making `Server` a case class and adding these methods to `Server` directly, which seemed really neat and tidy. I went with a separate builder class because

* code using twirp-runtime only deals with Builder classes and finagle `Service[Request, Response]`, so we're free to change the implementation of clients and servers around (=> Server is private now).
* this way, `Server` doesn't know anything about scalapb types
* it's more symmetric to how we build clients.